Hospital-System/HospitalSystem.BLL/Services/Appointment/DoctorTimeSlotsService.cs

```diff
@@ -0,0 +1,88 @@
+﻿using AutoMapper;
+using HospitalSystem.BLL.Services.Appointment.Interfaces;
+using HospitalSystem.Common.Models.DTOs.Appointment;
+using HospitalSystem.Common.Models.DTOs.Doctor;
+using HospitalSystem.DAL.Entities;
+using HospitalSystem.DAL.Repository.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.IdentityModel.Protocols;
+
+namespace HospitalSystem.BLL.Services.Appointment;
+
+public class DoctorTimeSlotsService : IDoctorTimeSlotsService
+{
+    private readonly IMapper _mapper;
+    private readonly IRepository<DoctorTimeSlot> _repository;
+
+    public DoctorTimeSlotsService(IMapper mapper, IRepository<DoctorTimeSlot> repository)
+    {
+        _mapper = mapper;
+        _repository = repository;
+    }
+
+    public async Task<bool> CreateAppointmentAsync(DoctorTimeSlotsDTO doctorTimeSlots, DateTime startsAt,
+        DateTime endsAt)
+    {
+        var isTimeSlotAvailable = await _repository.AnyAsync(x => x.StartsAt == startsAt && x.EndsAt == endsAt);
```

Comment:
What if time slot gonna be non standard? example 55 mins and you have predifined slots of 15 mins. Then u have one free slot and 3 busy what gonna be next?